cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -343,6 +344,7 @@ type NewAzCliArgs struct {
 	EnableTelemetry bool
 	// CommandRunner allows us to stub out the command execution for testing
 	CommandRunner exec.CommandRunner
+	HttpClient    httputil.HttpClient
```

Comment:
No need to change things now, but I wonder if longer term we should remove `httputil.HttpClient` in favor of just using `policy.Transporter` from the azcore package?  I think these interfaces try to solve the same problem,

Reply:
The interfaces are purposely the same so they align.  I just like the name `httputil.HttpClient` better because it better describes what it does.  What is the `policy.Transporter` interface going to do? Send me to another realm if a break a rule?